### ISSUE TITLE
Clarification of the required aria-level attribute

### ISSUE THREAD

**Issue by mfairchild365** [dequelabs/axe-core#814]
Just trying to understand this requirement.

The [ARIA 1.1 spec](https://www.w3.org/TR/wai-aria-1.1/#heading) says "Implicit Value for Role: Default for aria-level is 2." for the heading role.

The aria-level property suggests thats the user agent will automatically determine the value of level if it is not explicitly provided. "This attribute can be used to provide an explicit indication of the level when that is not possible to calculate from the document structure or the aria-owns attribute." Because the heading role defines a specific implicit default value, I would think that the heading role definition would override the general aria-level description.

Thus, my understanding was that it is okay to exclude the aria-level attribute. All elements that have the aria-role="heading" attribute will default to an aria-level of 2.

I've tested in VoiceOver, and the level defaults to 2, as I expected. I'm not sure about other screen readers.

Am I correct, or do I misunderstand the meaning of "implicit value"?

**Comment by WilcoFiers** [comment]
You're probably correct about this. We'd need to do some further testing in AT to make sure, but I was probably a little too hasty making that change in axe-core. @iamfaran did you have any thoughts on this? It was your ticket where we made that change.

**Comment by mfairchild365** [comment]
If I were to make a pull request to fix this, should I remove the rule entirely?

**Comment by iamrafan** [comment]
@WilcoFiers The [ARIA1.1](https://www.w3.org/TR/wai-aria-1.1/#heading) spec for ```role=heading``` says that ```aria-level``` is a required attribute and more over the AT experience on ```role=heading``` elements without the ```aria-level``` attribute is not consistent or reliable. 
Example: NVDA/FF treat all the role headings without an ```aria-level``` attribute as ```level-1``` headings

**Comment by mfairchild365** [comment]
Fair point about NVDA/FF defaulting to an aria-level of 1. That indicates to me that the 'implicit value' is not 'accessibility supported' per the WCAG standard and therefore this rule is helpful as is.

After reading the spec further I cam across this:

> 7.6 State and Property Attribute Processing
> However, required states and properties that are absent are an **author error**. Missing required states and properties are treated as if they were present and have an implicit neutral value that is not necessarily their default value. For example, the default value of aria-expanded is undefined, meaning neither expandable nor collapsible. But that does not apply to the case of a combobox. In this case, aria-expanded is needed to convey the expandable/collapsible nature of the combobox. Thus, the implicit value of aria-expanded for the combobox role is false, meaning expandable (and currently collapsed). The characteristics table associated with each WAI-ARIA role has an **"Implicit Value for Role" entry** that specifies the value of a state or property to use in the context of that role when the state or property is missing.

I interpret that to mean: If a role has an required attribute with a corresponding 'implicit value for role entry', AT should try to account for cases where the author failed to use the required attribute by using the implicit valu. Essentially smart error handling.

So, I agree with your conclusion @iamrafan *smile*

**Comment by iamrafan** [comment]
@mfairchild365 sure :) Have a nice day!
